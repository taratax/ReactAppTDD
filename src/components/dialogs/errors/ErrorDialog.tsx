import { createPortal } from "react-dom";

type ErrorDialogProps = {
  portalName: string;
  onClose: () => void;
};
export const ErrorDialog = (props: ErrorDialogProps) => {
  return createPortal(
    <>
      {/* <div ></div>
            <dialog open>
                <p>
                    Formularz zawiera błędy popraw je, a następnie ponownie wyślij formularz.
                </p>
                <button onClick={props.onClose}>OK</button>
            </dialog> */}

      <div className="fixed inset-0 bg-gray-900 opacity-40"></div>

      <div className="fixed inset-0 flex items-center justify-center">
        <div className="bg-white text-black">
        
            <dialog open>
              <p>
                Formularz zawiera błędy popraw je, a następnie ponownie wyślij
                formularz.
              </p>
              <button onClick={props.onClose}>OK</button>
            </dialog>
        
        </div>
      </div>
    </>,
    document.getElementById(props.portalName) as HTMLElement
  );

  // return (

  //     <>
  //     <div></div>
  //         <dialog open>
  //             <p>
  //                 Formularz zawiera błędy popraw je, a następnie ponownie wyślij formularz.
  //             </p>
  //             <button onClick={props.onClose}>OK</button>
  //         </dialog>

  //     </>
  // )
};
