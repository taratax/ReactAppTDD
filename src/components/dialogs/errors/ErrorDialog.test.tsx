import { ErrorDialog } from "./ErrorDialog";
import { render, screen } from "@testing-library/react";
import { portalNameDialogs } from "../../../constans/formFields";
import { createPortal } from "react-dom";

describe("The disalog should be visible", () => {


  it.skip("render skeleton of the modal dialog", () => {
    render(<ErrorDialog onClose={() => {}} portalName={portalNameDialogs} />);
    screen.logTestingPlaygroundURL();

    const theText = screen.getByText(
      /formularz zawiera błędy popraw je, a następnie ponownie wyślij formularz\./i
    );

    expect(theText).toBeInTheDocument();
  });

  it.skip("Render PopUp Properly", () => {
    const container = document.createElement("div");
    document.body.appendChild(container);
    createPortal(
      <ErrorDialog onClose={() => {}} portalName={portalNameDialogs} />,
      container
    );
    expect(container.innerHTML).toMatchSnapshot();
    expect(container).toBeInTheDocument();

    const theText = screen.getByText(
      /formularz zawiera błędy popraw je, a następnie ponownie wyślij formularz\./i
    );

    expect(theText).toBeInTheDocument();
  });
});
