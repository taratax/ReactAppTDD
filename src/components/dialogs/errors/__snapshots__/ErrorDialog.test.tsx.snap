// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The disalog should be visible Render PopUp Properly 1`] = `""`;
